packages/@uppy/dashboard/src/Dashboard.jsx

```diff
@@ -1015,10 +1017,34 @@ export default class Dashboard extends UIPlugin {
     })
   }
 
-  discoverProviderPlugins = () => {
-    this.uppy.iteratePlugins((plugin) => {
-      if (plugin && !plugin.target && plugin.opts && plugin.opts.target === this.constructor) {
-        this.addTarget(plugin)
+  // discoverProviderPlugins = () => {
+  //   this.uppy.iteratePlugins((plugin) => {
+  //     if (plugin && !plugin.target && plugin.opts && plugin.opts.target === this.constructor) {
+  //       console.log('plugin', plugin.id)
+  //       this.addTarget(plugin)
+  //     }
+  //   })
+  // }
+
+  #addSpecifiedPlugins = () => {
+    const plugins = this.opts.plugins || []
+
+    plugins.forEach((pluginID) => {
+      const plugin = this.uppy.getPlugin(pluginID)
+      if (plugin) {
+        plugin.mount(this, plugin)
+      }
+    })
+  }
+
+  #autoDiscoverPlugins = () => {
+    const typesAllowed = ['acquirer', 'editor']
```

Comment:
what's the purpose of this? could we add a comment?

Reply:
supported types? only these belong on the Dashboard, we wouldn’t want to try and mount Compressor or Tus.